### PR TITLE
Multisig tests

### PR DESCRIPTION
**Pull request by claudiu725**
- Add tests for the multisig smart contract which use the rust testing framework v2
- Add several `CodecFrom` implementations
- Introduce `CodecSelf` for allowing other types to override the default `CodecFrom` implementation
- Add `CodecFrom` implementations for `OptionalValue` to allow:
  - to use an `OptionalValue` whose inner type is compatible with the original type (implements `CodecFrom`)
  - to use a `value` directly instead of an `OptionalValue::Some(value)`
- Add `ManagedArgBuffer::into_multi_value_encoded` to allow using the proxy-generated argument buffer as a multi-value in order to forward the arguments to multisig

### CODE REVIEW

**Review by dorin-iancu — DISMISSED** [review]

File: contracts/examples/multisig/tests/multisig_rust_test_v2.rs [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+
+#[test]
+fn multisig_adder_test() {
+    let _ = DebugApi::dummy();
+
+    let world = &mut world();
+    let ic = &world.interpreter_context();
+
+    let users = setup_users(world, ic);
+    let source_adder = &mut adder_deploy(world, ic);
```
Comment (andrei-marinica) [inline contracts/examples/multisig/tests/multisig_rust_test_v2.rs:71]: `ic` is part of `world`, so I don't think we need it as a separate argument

File: contracts/examples/multisig/tests/multisig_rust_test_v2.rs [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+        world,
+        ic,
+    );
+    let expected_adder_address = AddressKey::interpret_from("sc:adder-multisig", ic);
+    assert_eq!(
+        adder.mandos_address_expr.value,
+        expected_adder_address.value
+    );
+
+    let first_value: BigUint = 42u64.into();
```
Comment (andrei-marinica) [inline contracts/examples/multisig/tests/multisig_rust_test_v2.rs:89]: Couldn't we actually only work with u64 or even u32 up to the contract proxy? There will unlikely be any big numbers in the test.

File: contracts/examples/multisig/tests/multisig_rust_test_v2.rs [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+) -> Option<Address> {
+    multisig_sign_multiple(action_id, signers, multisig, world);
+    multisig_perform(action_id, caller, multisig, world)
+}
+
+fn multisig_deploy_adder(
+    multisig: &mut MultisigContract,
+    source_adder: &mut AdderContract,
+    caller: &Address,
+    signers: &[&Address],
```
Comment (andrei-marinica) [inline contracts/examples/multisig/tests/multisig_rust_test_v2.rs:227]: Would probably be nice to group these arguments in a context object.

File: elrond-codec/src/equivalent/codec_from.rs [hunk 4]
```diff
@@ -7,7 +7,9 @@ where
 {
 }
 
-impl<T> CodecFrom<T> for T where T: TopEncodeMulti + TopDecodeMulti {}
+pub auto trait CodecSelf {}
```
Comment (andrei-marinica) [inline elrond-codec/src/equivalent/codec_from.rs:10]: Would rename to `CodecFromSelf`.

**Review by andrei-marinica — CHANGES_REQUESTED** [review]

**Review by andrei-marinica — APPROVED** [review]

**Review by andrei-marinica — APPROVED** [review]

**Pull request merged**